### ISSUE TITLE
Duplicated item loot pickups

### ISSUE THREAD

**Issue by ollisan05** [suriyun-production/mmorpg-kit-docs#673]
Hello I am fairly sure this is not a bug on your part of the kit but I don't know how to resolve this and people in the discord channel don't seem to know either, so im turning to you because It's really annoying, happens in both singleplayer and mmo server.

Here is a short video of what i mean: 
https://gyazo.com/751aa048e10d459915605fd4274827a1

**Comment by ollisan05** [comment]
Never mind this seems to be a problem with the latest version of the kit i tried it on a fresh new project latest update and it happens there too.

**Comment by Callepo** [comment]
![image](https://user-images.githubusercontent.com/7702658/98727809-d2965200-2398-11eb-887f-0d2db03d1bff.png)

Issue

**Comment by Callepo** [comment]
its adding items 2 times.. removed 1 and still got items .. idk if was intentional or what but yea.. that stops dups..

**Comment by insthync** [comment]
Function `IncreasingItemsWillOverwhelming` which has item amount list as one of its is working incorrectly. It is using a real list to add simulating items, it should create a simulating list before doing it.

Changes codes (from line 796):
```

    public static bool IncreasingItemsWillOverwhelming(this IList<CharacterItem> itemList, IEnumerable<ItemAmount> increasingItems, bool isLimitWeight, float weightLimit, float totalItemWeight, bool isLimitSlot, short slotLimit)
    {
        if (itemList == null || increasingItems == null)
            return false;
        List<CharacterItem> simulatingItemList = new List<CharacterItem>(itemList);
        foreach (ItemAmount receiveItem in increasingItems)
        {
            if (receiveItem.item == null || receiveItem.amount <= 0) continue;
            if (simulatingItemList.IncreasingItemsWillOverwhelming(
                receiveItem.item.DataId,
                receiveItem.amount,
                isLimitWeight,
                weightLimit,
                totalItemWeight,
                isLimitSlot,
                slotLimit))
            {
                // Overwhelming
                return true;
            }
            else
            {
                // Add item to temp list to check it will overwhelming or not later
                simulatingItemList.AddOrSetItems(CharacterItem.Create(receiveItem.item, 1, receiveItem.amount));
            }
        }
        return false;
    }

    public static bool IncreasingItemsWillOverwhelming(this IList<CharacterItem> itemList, IEnumerable<CharacterItem> increasingItems, bool isLimitWeight, float weightLimit, float totalItemWeight, bool isLimitSlot, short slotLimit)
    {
        if (itemList == null || increasingItems == null)
            return false;
        List<CharacterItem> simulatingItemList = new List<CharacterItem>(itemList);
        foreach (CharacterItem receiveItem in increasingItems)
        {
            if (receiveItem.IsEmptySlot()) continue;
            if (simulatingItemList.IncreasingItemsWillOverwhelming(
                receiveItem.dataId,
                receiveItem.amount,
                isLimitWeight,
                weightLimit,
                totalItemWeight,
                isLimitSlot,
                slotLimit))
            {
                // Overwhelming
                return true;
            }
            else
            {
                // Add item to temp list to check it will overwhelming or not later
                simulatingItemList.AddOrSetItems(CharacterItem.Create(receiveItem.dataId, receiveItem.level, receiveItem.amount));
            }
        }
        return false;
    }
```

**Comment by insthync** [comment]
Fixed in 1.60c
